compiler/test/compilable/test19295.d

```diff
@@ -0,0 +1,10 @@
+struct S1(T...) {
+    auto fun() {
+        static assert(__traits(compiles, &T[0]));
```

Comment:
This compiles now, but does the resulting delegate have the right `this` parameter?

Reply:
But the result is not a delegate. It's a function, so there is no this pointer.